hooks/useTimeAgo.tsx

```diff
@@ -0,0 +1,33 @@
+import { useCallback } from "react";
+import moment from "moment";
+
+const useTimeAgo = () => {
+  return useCallback((date: string) => {
+    const now = moment();
+    const createdMoment = moment(date);
+
+    const secondsThreshold = 60;
+    const minutesThreshold = 60;
+    const hoursThreshold = 24;
+    const daysThreshold = 7;
+
+    const diffInSeconds = now.diff(createdMoment, "seconds");
+    const diffInMinutes = now.diff(createdMoment, "minutes");
+    const diffInHours = now.diff(createdMoment, "hours");
+    const diffInDays = now.diff(createdMoment, "days");
+
+    if (diffInSeconds < secondsThreshold) {
+      return `${diffInSeconds} secs ago`;
+    } else if (diffInMinutes < minutesThreshold) {
+      return `${diffInMinutes} mins ago`;
+    } else if (diffInHours < hoursThreshold) {
+      return `${diffInHours} hours ago`;
+    } else if (diffInDays < daysThreshold) {
+      return `${diffInDays} days ago`;
+    } else {
+      return createdMoment.format("DD/MM/YY");
+    }
+  }, []);
+};
+
+export default useTimeAgo;
```

Comment:
This hook is unnecessary. You can use `momentjs` directly to achieve say `20 mins ago`.

`moment(date).fromNow()` or `moment([date]).fromNow()`

reference [https://momentjs.com/docs/#/displaying/fromnow/](url)

Please delete this hook file

Reply:
I'll delete the hook, however, there's a limitation to this `moment(date).fromNow()`, because in the design, there's a scenario where the token is quite dated, and the date is rendered in this format `DD/MM/YY`. So, this `moment(date).fromNow()` won't suffice.
I'll do this instead

`
const timeAgo = (date) => {
  const now = moment();
  const createdMoment = moment(date);
  const daysThreshold = 7;

  const diffInDays = now.diff(createdMoment, "days");

  if (diffInDays >= daysThreshold) {
    return createdMoment.format("DD/MM/YY");
  } else {
    return createdMoment.fromNow();
  }
};
`